### PR TITLE
Breaking down examples to plotting and manipulating and advanced

### LINKED ISSUE #884: Better breakdown of the examples
We are starting to have a lot of examples, which is a good thing, but can easily lead to getting lost in the information.

I suggest a new breakdown of examples:

* General: just a handful of examples

* Decoding: what we currently have

* Functional connectivity: what we currently have

* Plotting

* Manipulation of brain images

* Advanced

Basically, it would entail breaking up our current 'visualizing_and_manipulating' into 'plotting', 'manipulating' and 'advanced'. Advanced is just another word for 'other', in a certain sense. We would put things like the low-level ICA stuff, or the mass-univariate stuff. Ie things that we don't really want to advertise strong to end users.

One strong reason for me to separate out the plotting examples is that it seems that our users do not use the full breadth of the plotting options, and do not find them.

This is an issue labelled 'discussion', please do comment here on what you think about this plan.

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Should fix #884

### CODE REVIEW

**Comment by lesteve** [comment]
From a quick glance at it, LGTM.

A few comments:
* the issue was talking about moving around the examples but you moved also some of the documentation, e.g. plotting.rst got moved from doc/manipulating_visualizing to doc/plotting manipulating_images.rst got moved from doc/manipulating_visualizing to manipulating_images. Not sure whether the generated HTML looks any different. If it doesn't that's probably fine.
* Someone should double-check once CircleCI that the breakdown of the examples by category makes sense. In particular what went into the "Advanced" section. Here is a comment from @GaelVaroquaux from the associated issue in case it helps:
> Advanced is just another word for 'other', in a certain sense. We would put things like the low-level ICA stuff, or the mass-univariate stuff. Ie things that we don't really want to advertise strong to end users.

**Comment by lesteve** [comment]
Looks like CircleCI fails with a genuine sphinx warning.

```
SphinxWarning: /home/ubuntu/nilearn/doc/plotting/plotting.rst:5: WARNING: duplicate label plotting, other instance in /home/ubuntu/nilearn/doc/plotting/index.rst
```

**Comment by KamalakerDadi** [comment]
moved two mass-univariate examples from manipulating into advanced section. Did I miss something more here ?

**Comment by lesteve** [comment]
> Did I miss something more here ?

@GaelVaroquaux mentions the "low-level ICA stuff". Not sure which one it is BTW. @AlexandreAbraham may know.

**Comment by KamalakerDadi** [comment]
> Looks like CircleCI fails with a genuine sphinx warning.

I think there is a naming conflict, pushed a recent commit to see if it passes. If I am not wrong.

**Comment by KamalakerDadi** [comment]
> low-level ICA stuff

I have no clear example to move. May be I thought this ```plot_ica_resting_state.py``` but I am not sure.

**Comment by lesteve** [comment]
> I think there is a naming conflict, pushed a recent commit to see if it passes. If I am not wrong.

Just in case you didn't know, you can check on your machine using `make html-strict` in the doc folder.

**Comment by KamalakerDadi** [comment]
> you can check on your machine using make html-strict

I checked it, this time got an latex error which is not related to my PR.

File: doc/plotting/index.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+neuroimaging volumes with nilearn.
+
+|
+
+.. include:: ../includes/big_toc_css.rst
+
+
+.. toctree::
+
+   plotting.rst
```
Comment (GaelVaroquaux) [inline doc/plotting/index.rst]: I am not sure that this extra level of indirection is useful. Why don't you just include the contents of plotting.rst here (and thus remove the different tune_toc doc)?

File: doc/manipulating_images/index.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. include:: ../tune_toc.rst
+
+
+.. _manipulating_images:
```
Comment (GaelVaroquaux) [inline doc/manipulating_images/index.rst]: I would call this link label "image_manipulation", to be closer to the title that it represents.

**Comment by GaelVaroquaux** [comment]
Overall, this looks great.

I would like to change the ordering of the examples in the documentation. For this I suggest renaming the directors from "foobar" to "01_foobar".

The ordering that I would like is:

* 01_plotting
* 02_decoding
* 03_connectivity
* 04_manipulating_images
* 05_advanced

File: doc/manipulating_images/index.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+=================================
+Image manipulation
+=================================
+
+In this section, we detail the general tools to manipulation of
```
Comment (banilo) [inline doc/manipulating_images/index.rst]: simplify "tools to manipulate"

**Comment by GaelVaroquaux** [comment]
>     Here is a comment from @GaelVaroquaux from the associated issue in
>     case it helps: > Advanced is just another word for 'other', in a
>     certain sense. We would put things like the low-level ICA stuff, or
>     the mass-univariate stuff. Ie things that we don't really want to
>     advertise strong to end users.

Yes, I agree that we should move the low-level ICA stuff there. That's a
good idea.

**Comment by GaelVaroquaux** [comment]
>     low-level ICA stuff

> I have no clear example to move. May be I thought this
> plot_ica_resting_state.py but I am not sure.

Yes, that one!

File: doc/plotting/index.rst (hunk 1 above)
Comment (KamalakerDadi) [inline doc/plotting/index.rst]: To make sure myself, this means that should move content from plotting.rst here and remove both name and file plotting.rst and tune_toc.rst ?

Comment (GaelVaroquaux) [inline doc/plotting/index.rst]: > To make sure myself, this means that should move content from plotting.rst here and remove both name and file plotting.rst and tune_toc.rst ?

Correct for the first part, but don't remove tune_toc.rst, as it's used
elsewhere.

File: doc/manipulating_images/index.rst (hunk 3 above)
Comment (KamalakerDadi) [inline doc/manipulating_images/index.rst]: > For this I suggest renaming the directors from "foobar" to "01_foobar".

I am not sure which director ? It is folder name you mean ?

Comment (GaelVaroquaux) [inline doc/manipulating_images/index.rst]: >     For this I suggest renaming the directors from "foobar" to "01_foobar".

> I am not sure which director ? It is folder name you mean ?

Sorry: typo: I meant "directories".

**Comment by KamalakerDadi** [comment]
@GaelVaroquaux Could you please have a review if I have done according to your comments ?

**Comment by GaelVaroquaux** [comment]
Tonight; Right now I am in a meeting for a few hours.

**Comment by banilo** [comment]
LGTM

**Comment by lesteve** [comment]
Can someone with more domain expertise than me quickly check visually the gallery? https://circle-artifacts.com/gh/nilearn/nilearn/998/artifacts/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/index.html.

**Comment by lesteve** [comment]
> I would like to change the ordering of the examples in the documentation. For this I suggest renaming the directors from "foobar" to "01_foobar".

> The ordering that I would like is:

>    01_plotting
    02_decoding
    03_connectivity
    04_manipulating_images
    05_advanced

BTW @Titan-C at one point maybe we could make the ordering of the subsections configurable in sphinx-gallery (I guess right now it is alphabetical) so that we don't have to resort to renaming folders like this.

**Comment by GaelVaroquaux** [comment]
> BTW @Titan-C at one point maybe we could make the ordering of the subsections
> configurable in sphinx-gallery (I guess right now it is alphabetical) so that
> we don't have to resort to renaming folders like this.

+1

**Comment by GaelVaroquaux** [comment]
@KamalakerDadi : I am really sorry, I didn't explain things properly. I would like this directory change to happen only in the examples, not in the documentation. There is no need to have the numbers hard wired in the directories of the documentation.

**Comment by GaelVaroquaux** [comment]
Aside from that comment, the PR looks great.

**Comment by KamalakerDadi** [comment]
> Aside from that comment, the PR looks great.

I will finish it soon. I will get back if I have any rendering issues.

**Comment by bthirion** [comment]
The result looks good. As a by-product, it shows that some examples should be renamed, but this should be done in another PR.

**Comment by GaelVaroquaux** [comment]
CircleCI is failing because there is an error building the docs: it's not visible on the main page, but scrolling down the logs in artifacts (https://circle-artifacts.com/gh/nilearn/nilearn/1028/artifacts/0/home/ubuntu/log.txt) reveals it:

<pre>

Traceback (most recent call last):
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/application.py", line 220, in build
    self.builder.build_update()
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' % len(to_build))
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 276, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 320, in write
    self._write_serial(sorted(docnames), warnings)
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 335, in _write_serial
    self.warn(*warning)
  File "/home/ubuntu/virtualenvs/venv-system/local/lib/python2.7/site-packages/sphinx/application.py", line 268, in warn
    raise SphinxWarning(warntext)
SphinxWarning: /home/ubuntu/nilearn/doc/decoding/decoding_tutorial.rst:322: WARNING: undefined label: sphx_glr_auto_examples_decoding_plot_haxby_full_analysis.py (if the link has no caption the label must precede a section header)


Warning, treated as error:
/home/ubuntu/nilearn/doc/decoding/decoding_tutorial.rst:322: WARNING: undefined label: sphx_glr_auto_examples_decoding_plot_haxby_full_analysis.py (if the link has no caption the label must precede a section header)
</pre>

**Comment by KamalakerDadi** [comment]
> it's not visible on the main page, but scrolling down the logs in artifacts

@GaelVaroquaux Sorry, I should have seen it myself. But, its done now.

> As a by-product, it shows that some examples should be renamed, but this should be done in another PR

@bthirion Do you want me to open an issue ? Could you give which examples in particular ?

**Comment by GaelVaroquaux** [comment]
That's great. I'm merging. Thank you!

**Pull request merged**